### PR TITLE
feat: import wallet

### LINKED ISSUE #137: Import wallet using seedphrase
Existing power users should be able to move their wallets to this UI

### PR DESCRIPTION
**Pull request by theborakompanioni**
Closes #137.

Adds the ability to import an existing wallet in Jam.

- [x] Customize options (plain input of gaplimit and blockheight)
- [x] Reset `gaplimit` to original value
- [x] Add "back" button
- [x] Prevent importing if rescan is active
- [x] Call `/display` after rescanning finished
- [x] Test on `mainnet`

## How to test
After checking out the branch, execute `npm run regtest:rebuild` in order to rebuild your dev environment to get the lastest upstream changes from JM (the "import wallet" endpoint is only on `master` and not released yet). Otherwise you will get `"Error while importing the wallet. Reason: Method Not Allowed"` when trying to import a wallet.

### CODE REVIEW

**Comment by theborakompanioni** [comment]
Hey @editwentyone

This is not functional and can be tested. 
From my point of view the only thing that is missing the the blockheight-as-date-picker component (Currently, it is a normal number input). If that is okay with you, I'll tackle this in a follow-up PR.

What is still missing but needed before merging:
- Show options only with jm backenv version 0.9.10 (or 0.9.10dev) or greater
- Call /display after rescanning finished
- Improve or hide "rescan" page
- Test on `mainnet`

Also, this has not been tested on `mainnet` yet.

Any feedback to the current state would be highly appreciated :pray:

**Comment by theborakompanioni** [comment]
Follow-up PRs for stuff noticed during development:
- [ ] blockheight-as-date-picker component in Import Details Form
- [ ] Refactor: Consistently use `walletName` or/and `walletFileName`
- [ ] Feat: Validate world list with BIP39
- [ ] Feat: Check for existing wallet before importing/creating wallets

File: src/i18n/locales/en/translation.json
Comment (openoms) [inline src/i18n/locales/en/translation.json:168]: Despite the usual definition the gaplimit setting for JoinMarket needs to be the highest number of address used in any mixdepth.

Attempting to improve the wording in the Jam context:
```suggestion
      "description_gaplimit": "Set to the highest address number used in any of the jars. Increase this number if your wallet is heavily used.",
```

Comment (openoms) [reply src/i18n/locales/en/translation.json]: Agree. Let's not try to redefine the gaplimit used in other software.
How about "address height"?

**Comment by theborakompanioni** [comment]
Rebased on master.

This is now "complete" and is ready for review.
Suggestions for the wording (in translation file) would be highly appreciated. But of course, the proper functioning of the new feature should be the main focus. :pray: 

I will create issues for following related tasks which will be done in follow-up PRs:
- [ ] Show options only with jm backenv version 0.9.10 (or 0.9.10dev) or greater
- [ ] Input custom rescan blockheight as year/month date picker

As well as for additional tasks that came up during development:
- [ ] Feat: Validate mnemonic phrase (display a warning if a word is not contained in BIP39, but let user submit the form)
- [ ] Refactor: Consistently use `walletName` or/and `walletFileName`
- [ ] Refactor: use `reloadWalletInfo` from `App` component consistently

**Comment by climardo** [comment]
👏 congrats. ive been following this pr since i had to do this manually and ive been looking forward to this. going to try and run this to give feedback (im a newb)

**Comment by theborakompanioni** [comment]
> 👏 congrats. ive been following this pr since i had to do this manually and ive been looking forward to this. going to try and run this to give feedback (im a newb)

This is so nice! ANY feedback is highly welcome! You can start with [setting up the regtest environment](https://github.com/joinmarket-webui/jam/blob/master/docker/regtest/readme.md)! If you need any help or assistance, just ping me! :pray:

**Comment by theborakompanioni** [comment]
Rebased on master (7a6b920fd2f145b9e47f29ef0f27a2786bb87525)

**Review by editwentyone — APPROVED** [review]
cACK, on testnet, not main net

**Comment by theborakompanioni** [comment]
With help of some users, this has been tested on `mainnet` with a moderately used wallets. Rescanning from block `481824` (segwit activation) took around ~8 hours on average (on a raspberry).  Wallet has been successfully imported. However, tests were always done on the Bitcoin Core instance, that was already aware of the wallet. It should work on pristine Bitcoin Core instances as well, but we have to wait for feedback to verify.

**Pull request merged**
